controllers/buildpipeline/buildpipeline_adapter.go

```diff
@@ -132,6 +133,10 @@ func (a *Adapter) EnsureSnapshotExists() (controller.OperationResult, error) {
 	err = a.client.Create(a.context, expectedSnapshot)
 	if err != nil {
 		a.logger.Error(err, "Failed to create Snapshot")
+		if errors.IsForbidden(err) {
+			// we cannot create a snapshot and we don't want to block resources, user has to retry
```

Comment:
nit: can we specify an example (like: we hit the snapshot quota) of why we weren't able to create a snapshot? Because we are requeuing for other kinds of errors
```suggestion
			// we cannot create a snapshot (possibly because the snapshot quota is hit) and we don't want to block resources, user has to retry
```